Running tests; contributing instructions

I was wondering what is the expected way to run tests. I cannot find any documentation about it.

After installing loky with `pip install -e .`, running
```
pytest .
```
results in 3 test failures in `tests/test_loky_backend.py` (on Linux, Python 3.6), for instance
```py
______________________________________________ TestLokyBackend.test_interactively_define_process_no_main[True] ______________________________________________

self = <tests.test_loky_backend.TestLokyBackend object at 0x7efc6f0aa2e8>, run_file = True
[...]
E               ValueError: Non-zero return code: 1.
E               Stdout:
E               
E               Stderr:
E               Traceback (most recent call last):
E                 File "/tmp/tmp5c1ijwmn_joblib.py", line 1, in <module>
E                   from loky.backend.process import LokyProcess
E               ImportError: No module named 'loky'

tests/utils.py:111: ValueError
```
the remaining 212 tests pass. 

I can see that Travis CI uses tox [that does call pytest](https://github.com/tomMoral/loky/blob/master/tox.ini#L36) but shouldn't this also work without tox?

OK, @massich  the issue was that I forgot to install pytest in the env, so it was taking the system pytest. 

Installing pytest, fixes the above test failures. But now I get another one,

```py
TestLokyBackend.test_sync_object_handling _________________________________________________________

self = <tests.test_loky_backend.TestLokyBackend object at 0x7f564ea42ba8>

    def test_sync_object_handling(self):
        """Check the correct handling of semaphores and pipes with loky
    
            We use a Pipe object to check the stated of file descriptors in parent
            and child. To make sure there is no interference in the fd numbers, we
            use high number fd, so newly created fd should be inferior.
    
            To ensure we have the right number of fd in the child Process, we used
            `lsof` as it is compatible with Unix systems.
            Different behaviors are observed with the open fds, in particular:
            - python2.7 and 3.4 have an open fd for /dev/urandom.
            - python2.7 links stdin to /dev/null even if it is closed beforehand.
            """
    
        # TODO generate high numbered multiprocessing.Pipe directly
        # -> can be used on windows
        r, w = self._high_number_Pipe()
    
        tmp_fname = "/tmp/foobar" if sys.platform != "win32" else ".foobar"
        with open(tmp_fname, "w"):
            # Process creating semaphore and pipes before stopping
            started, stop = self.Event(), self.Event()
            p = self.Process(target=self._test_sync_object_handling,
                             args=(started, stop, r, w.fileno()))
            named_sem = []
            try:
    
                p.start()
                assert started.wait(5), "The process took too long to start"
                r.close()
                w.send_bytes(b"foo")
                if sys.platform != "win32":
>                   named_sem = self._check_fds(p.pid, w.fileno())

tests/test_loky_backend.py:555: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <tests.test_loky_backend.TestLokyBackend object at 0x7f564ea42ba8>, pid = 357, w = 115

    def _check_fds(self, pid, w):
        """List all the open files and check no extra files are presents.
    
            Return a list of open named semaphores
            """
        import subprocess
        try:
            out = subprocess.check_output(["lsof", "-a", "-Fftn",
                                           "-p", "{}".format(pid),
                                           "-d", "^txt,^cwd,^rtd"])
            lines = out.decode().split("\n")[1:-1]
        except FileNotFoundError:
            print("lsof does not exist on this plateform. Skip open files"
                  "check.")
            return []
    
        n_pipe = 0
        named_sem = []
        for fd, t, name in zip(lines[::3], lines[1::3], lines[2::3]):
    
            # Check if fd is a standard IO file. For python2.7, stdin is set
            # to /dev/null during `Process._boostrap`. For other version, stdin
            # should be closed.
            is_std = (fd in ["f1", "f2"])
            if sys.version_info[:2] < (3, 3):
                if sys.platform != "darwin":
                    is_std |= (fd == "f0" and name == "n/dev/null")
                else:
                    is_std |= (name == "n/dev/null")
    
            # Check if fd is a pipe
            is_pipe = (t in ["tPIPE", "tFIFO"])
            n_pipe += is_pipe
    
            # Check if fd is open for the rng. This can happen on different
            # plateform and depending of the python version.
            is_rng = (name == "n/dev/urandom")
    
            # Check if fd is a semaphore or an open library. Store all the
            # named semaphore
            is_mem = (fd in ["fmem", "fDEL"])
            if sys.platform == "darwin":
                is_mem |= "n/loky-" in name
            if is_mem and "n/dev/shm/sem." in name:
                named_sem += [name[1:]]
    
            # no other files should be opened at this stage in the process
>           assert (is_pipe or is_std or is_rng or is_mem)
E           assert (False or False or False or False)

tests/test_loky_backend.py:510: AssertionError
```

--- a/README.md
+++ b/README.md
@@ -62,6 +62,27 @@ print("Number of used processes:", n_workers)
 assert n_workers == 4
 ```
 
+### Workflow to contribute
+
+To contribute to **loky**, first create an account on [github](http://github.com/).
+Once this is done, fork the [loky repository](http://github.com/loky/loky) to
+have your own repository, clone it using 'git clone' on the computers where you
+want to work. Make your changes in your clone, push them to your github account,
+test them on several computers, and when you are happy with them, send a pull
+request to the main repository.
+
+### Running the test suite
+
+To run the test suite, you need the `pytest` (version >= 3) and `psutil`
+modules. Run the test suite using:
+
+```sh
+    pip install -e .
+    pytest .
+```
+
+from the root of the project.
+
 ### Acknowledgement
 
 This work is supported by the Center for Data Science, funded by the IDEX

--- a/setup.py
+++ b/setup.py
@@ -82,4 +82,6 @@ setup(
     cmdclass=cmdclass,
     platforms='any',
     install_requires=['cloudpickle'],
+    tests_require = ['pytest', 'psutils',],
+
 )

